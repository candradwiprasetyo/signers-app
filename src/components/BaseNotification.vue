<template>
  <div
    v-if="show"
    :class="[
      'fixed bottom-6 right-6 z-50 px-4 py-3 rounded shadow-lg text-white transition-all duration-300',
      type === 'success' ? 'bg-green-500' : 'bg-red-500',
    ]"
  >
    {{ message }}
  </div>
</template>

<script setup lang="ts">
defineProps<{
  show: boolean
  message: string
  type: 'success' | 'error'
}>()
</script>
