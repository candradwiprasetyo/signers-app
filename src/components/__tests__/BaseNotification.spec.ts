import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import BaseNotification from '@/components/BaseNotification.vue'

describe('BaseNotification', () => {
  it('render success message correctly', () => {
    const wrapper = mount(BaseNotification, {
      props: {
        show: true,
        message: 'Data berhasil disimpan',
        type: 'success',
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('render error message correctly', () => {
    const wrapper = mount(BaseNotification, {
      props: {
        show: true,
        message: 'Terjadi kesalahan saat menyimpan data',
        type: 'error',
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
