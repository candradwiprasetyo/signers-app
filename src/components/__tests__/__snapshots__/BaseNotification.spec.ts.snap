// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseNotification > render error message correctly 1`] = `"<div class="fixed bottom-6 right-6 z-50 px-4 py-3 rounded shadow-lg text-white transition-all duration-300 bg-red-500">Terjadi kesalahan saat menyimpan data</div>"`;

exports[`BaseNotification > render success message correctly 1`] = `"<div class="fixed bottom-6 right-6 z-50 px-4 py-3 rounded shadow-lg text-white transition-all duration-300 bg-green-500">Data berhasil disimpan</div>"`;
